### PR TITLE
Refactor NewInformer Functions

### PR DESCRIPTION
**Pull request by hongchaodeng**
NewInformer and NewIndexerInformer are using the same code.
Combined them here.

### CODE REVIEW

File: pkg/controller/framework/controller.go [hunk 1]
```diff
@@ -284,7 +246,17 @@ func NewIndexerInformer(
 ) (cache.Indexer, *Controller) {
 	// This will hold the client state, as we know it.
 	clientState := cache.NewIndexer(DeletionHandlingMetaNamespaceKeyFunc, indexers)
+	informer := newInformer(lw, objType, resyncPeriod, h, clientState)
+	return clientState, informer
+}
 
+func newInformer(
```
Comment (xiang90) [inline pkg/controller/framework/controller.go]: one line?

**Comment by xiang90** [comment]
looks good.

**Comment by derekwaynecarr** [comment]
LGTM

File: pkg/controller/framework/controller.go (hunk 1 above)
Comment (lavalamp) [inline pkg/controller/framework/controller.go]: I think it's readable like this :)

**Comment by lavalamp** [comment]
LGTM-- this has been on my list for ages. Eventually I intend to let callers pass in a store...

**Comment by hongchaodeng** [comment]
Closing this in favor of #27078

**Pull request closed without merging**